charts/prometheus/values.schema.json

```diff
@@ -271,6 +271,9 @@
                 "global": {
                     "type": "object",
                     "properties": {
+                        "external_labels": {
+                           "type": "object"
+                        },
```

Comment:
As shown above, you are setting the field nested in `server`, not in `global`. The schema has to include `server.externalLabels` instead. Please, remember to insert the new field in `values.yaml` as well with a short comment/reference as seen e.g. at `server.remoteRead`.